backend/main/src/Group/Repositories/ContainerRepository.ts

```diff
@@ -0,0 +1,195 @@
+	async find(id: ContainerId[]): Promise<Container[]>;
+	async find(
+		id: ContainerId | ContainerId[],
+	): Promise<Container | Container[] | null> {
+		// if the ID is array.
+		if (Array.isArray(id)) {
+			const containers = await Promise.all(
+				id.map((containerId) =>
+					this.nishikiDbClient.getContainer(containerId.id),
+				),
+			);
+			const containersObject: Container[] = [];
+
+			for (const container of containers) {
+				if (container) {
+					const containerObject = createContainerObject(container);
+					containersObject.push(containerObject);
+				}
+			}
+
+			return containersObject;
+		}
+
+		const containerData = await this.nishikiDbClient.getContainer(id.id);
+
+		return containerData ? createContainerObject(containerData) : null;
+	}
+
+	/**
+	 *
+	 * Create a container.
+	 * @param container
+	 */
+	async create(container: Container): Promise<undefined> {
+		await this.nishikiDbClient.saveContainer({
+			containerId: container.id.id,
+			containerName: container.name,
+			foods: container.foods.map((food) => fromFoodObjectToFoodData(food)),
+		});
+	}
+
+	/**
+	 * Update a container
+	 * @param container
+	 */
+	async update(container: Container): Promise<undefined> {
+		await this.nishikiDbClient.saveContainer({
+			containerId: container.id.id,
+			containerName: container.name,
+			foods: container.foods.map((food) => fromFoodObjectToFoodData(food)),
+		});
+	}
+
+	/**
+	 * Delete a container
+	 * @param id - target container id
+	 */
+	async delete(id: ContainerId): Promise<undefined> {
+		await this.nishikiDbClient.deleteContainer(id.id);
+	}
+}
+
+const fromFoodObjectToFoodData = (food: Food): FoodData => {
+	return {
+		foodId: food.id.id,
+		name: food.name,
+		quantity: food.quantity?.quantity ?? null,
+		unit: food.unit?.name ?? null,
+		expiry: food.expiry?.date.toISOString() ?? null,
+		category: food.category,
+		createdAt: food.createdAt.toISOString(),
+	};
+};
+
+/**
+ * Create a container from primitive values.
+ * When containerData is invalid, this function throws an error.
+ * @param containerData
+ * @throws ContainerRepositoryError - this error will be thrown when the data is invalid.
+ */
+const createContainerObject = (containerData: ContainerData): Container => {
+	const containerIdOrErr = ContainerId.create(containerData.containerId);
+	const name = containerData.containerName;
+	const foods = containerData.foods.map((food) => {
+		const foodIdOrErr = ContainerId.create(food.foodId);
+		if (!foodIdOrErr.ok) {
+			const report = `ContainerId: ${containerData.containerId}, errorResult.error.message`;
+			throw new ContainerRepositoryError(foodIdOrErr.error.message, report);
+		}
+
+		const foodId = foodIdOrErr.unwrap();
+		const name = food.name;
+		const foodProps: IFoodProps = {
+			name: name,
+			category: food.category,
+			createdAt: new Date(food.createdAt),
+		};
+
+		if (food.quantity) {
+			const quantityOrError = Quantity.create(food.quantity);
+			if (!quantityOrError.ok) {
```

Comment:
You can use the err property. Using this, this can be easier to understand.
This is optional.

```suggestion
			if (quantityOrError.err) {
```